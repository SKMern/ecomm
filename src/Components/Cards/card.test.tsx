// Generated by CodiumAI
import React from 'react'
import { render } from "@testing-library/react";
import ImageCard from ".";

describe('ImageCard', () => {

    // Tests that ImageCard component renders correctly with valid props
    it('should render correctly with valid props', () => {
        const props = {
            category: 'electronics',
            description: 'Lorem ipsum dolor sit amet',
            id: 1,
            _id: '1234',
            image: 'https://example.com/image.jpg',
            price: 100,
            rating: { rate: 4.5, count: 10 },
            title: 'Product Title'
        };
        const wrapper = render(<ImageCard {...props} />);
        expect(wrapper).toMatchSnapshot();
    });


});
