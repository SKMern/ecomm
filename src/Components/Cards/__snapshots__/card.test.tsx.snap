// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCard should render correctly with valid props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-14yw00i-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardMedia-root css-1999db7-MuiCardMedia-root"
          role="img"
          style="background-image: url(https://example.com/image.jpg);"
          title="Product Title"
        />
        <div
          class="MuiCardContent-root css-15p27lc-MuiCardContent-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom MuiTypography-noWrap css-fcsi4c-MuiTypography-root"
          >
            Product Title
          </div>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom MuiTypography-noWrap css-e3y5cv-MuiTypography-root"
          >
            Lorem ipsum dolor sit amet
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Rs. 
            100
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-14yw00i-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardMedia-root css-1999db7-MuiCardMedia-root"
        role="img"
        style="background-image: url(https://example.com/image.jpg);"
        title="Product Title"
      />
      <div
        class="MuiCardContent-root css-15p27lc-MuiCardContent-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom MuiTypography-noWrap css-fcsi4c-MuiTypography-root"
        >
          Product Title
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom MuiTypography-noWrap css-e3y5cv-MuiTypography-root"
        >
          Lorem ipsum dolor sit amet
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Rs. 
          100
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
